channel/channel.go

```diff
@@ -321,28 +328,37 @@ func (c *Channel) SignAndAddState(s state.State, sk *[]byte) (state.SignedState,
 
 // UpdateWithChainEvent mutates the receiver with the supplied chain event, replacing the relevant data fields.
 func (c *Channel) UpdateWithChainEvent(event chainservice.Event) (*Channel, error) {
-	switch e := event.(type) {
-	case chainservice.AllocationUpdatedEvent:
-		c.OnChain.Holdings[e.AssetAddress] = e.AssetAmount
-		// TODO: update OnChain.StateHash and OnChain.Outcome
-	case chainservice.DepositedEvent:
-		c.OnChain.Holdings[e.Asset] = e.NowHeld
-	case chainservice.ConcludedEvent:
-		break // TODO: update OnChain.StateHash and OnChain.Outcome
-	case chainservice.ChallengeRegisteredEvent:
-		h, err := e.StateHash(c.FixedPart)
-		if err != nil {
-			return nil, err
+	if event.BlockNum() > c.LastChainUpdate.BlockNum ||
+		(event.BlockNum() == c.LastChainUpdate.BlockNum && event.TxIndex() > c.LastChainUpdate.TxIndex) {
```

Comment:
Made a suggestion here #1675

Reply:
That make sense. Merged that PR into this branch